### DISCUSSION TITLE
Questions on using Djot

### DISCUSSION THREAD

**Discussion by jodros** [Omikhleia/markdown.sile#131] [Q&A] [answered]
First time trying Djot out and having some little troubles...

- Can't break pages, {.pagebreak} just doesn't work. (for now I'm doing it with  `\pagebreak`{=sile})
- According to the [official documentation](https://djot.net/syntax#comment), comments are made within blocks of `%`, but in my file they're are being actually printed!
- I didn't find in the manual how to change the font size (among other font properties) of a text block
- And speaking of text blocks, I thought that in blocks with colons like  ::: :::, the colons weren't printed, but they are, I'm not sure  about how to use them...
- `## Heading {#markdown-unnumbered-heading .unnumbered}` is not suppressing the numbering...

**Comment by Omikhleia** [comment, marked as answer]
> Can't break pages, {.pagebreak} just doesn't work. 

How or where did you use it? My implementation only support it on "thematic breaks".
But I use it a lot without issues; e.g. from one of my books:

```
{.none .pagebreak}
---
```

(The `.none` here depends on whether you want a visible thematic break or not, before the page break)

> According to the official documentation, comments are made within blocks of %, but (...)

Indeed, comments in Djot are somewhat broken. This might have been improved in the JS implementation while the Lua implementation is still behind, I haven't checked it. Still, there's something unclear about comment blocks (discussed in one of the Djot issues, IIRC) so I wouldn't recommend using them at this point.

> `## Heading {#markdown-unnumbered-heading .unnumbered}` is not suppressing the numbering

The above is not a valid Djot block syntax... it's a Markdown one :smile_cat: 
Djot tries to remove the ambiguity on block attributes vs. inline attributes.

Here Djot would mandate _block_ attributes to be used, on the previous line(s) before the block:

```
{#my-heading-label .unnumbered}
## My Heading Title
```

Otherwise, Djot regards them as inline attributes, i.e. applying to the _inline_ content (and here, you have a space, I think those attribues are "unattached" in that case, but e.g. `...Title{.attributes}` would apply them to the "Title" text).

> I didn't find in the manual how to change the font size (among other font properties) of a text block

There are multiple solutions - but I'll assume here that you use my resilient book class, and I'll then recommend styling in that case, for instance:

- In your Djot file, `{custom-style=MyJodrosStyle}` (before a block or just after an inline element)
- In the style file, add the custom style, and tune it at convenience:

   ```
   MyJodrosStyle:
     style:
       font:
         size: "0.9em" # or whatever size you wanted...
   ```

**Reply by Omikhleia** [reply]
( @jodros I moved the original issue to Q&A discussions )
